pkg/defaults/defaults.go

```diff
@@ -144,7 +145,7 @@ func FromConfig(
 			releaseStep = release.AssembleReleaseStep(*rawStep.ReleaseImagesTagStepConfiguration, config.Resources, podClient, imageClient, artifactDir, jobSpec)
 
 		} else if rawStep.TestStepConfiguration != nil {
-			step = steps.TestStep(*rawStep.TestStepConfiguration, config.Resources, podClient, artifactDir, jobSpec)
+			step = steps.TestStep(*rawStep.TestStepConfiguration, config.Resources, podClient, artifactDir, jobSpec, testSecret)
```

Comment:
It also needs to be validated. see [here](https://github.com/openshift/ci-operator/blob/master/pkg/api/config.go#L90-L112)

Reply:
So you want this to be configurable instead of flag, or both?

Reply:
I would expect it to be defined in [here](https://github.com/openshift/ci-operator/blob/master/pkg/defaults/defaults.go#L409)

Reply:
Of course, if we do this like this we won't have to make any updates in prowgen, since ci-operator will read the secret directly from its config. 

Let us decide then. @petr-muller @bbguimaraes @stevekuznetsov  Any thoughts on this one?

Reply:
Which make sense, as now you can define secret only for steps you need, and not for all of them. Limits attach surface.